### PR TITLE
[AOT] added AotCompatibility test app exit code assertion

### PR DESCRIPTION
**Pull request by Yun-Ting**
Fixes #
Design discussion issue #

## Changes

Please provide a brief description of the changes here.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: build/assert-publish-aot-testApp-warning-count.ps1 [hunk 1]
```diff
@@ -12,6 +12,21 @@ foreach ($line in $($publishOutput -split "`r`n"))
     }
 }
 
+pushd $rootDirectory\test\OpenTelemetry.AotCompatibility.TestApp/bin/Debug/net7.0/linux-x64
+chmod +x ./OpenTelemetry.AotCompatibility.TestApp
+chmod 777 ./OpenTelemetry.AotCompatibility.TestApp
```
Comment (eerhardt) [inline build/assert-publish-aot-testApp-warning-count.ps1:17]: This shouldn't be necessary. Why is this needed?

File: build/assert-publish-aot-testApp-warning-count.ps1 [hunk 2]
```diff
@@ -12,6 +12,19 @@ foreach ($line in $($publishOutput -split "`r`n"))
     }
 }
 
+pushd $rootDirectory\test\OpenTelemetry.AotCompatibility.TestApp/bin/Debug/net7.0/linux-x64
```
Comment (eerhardt) [inline build/assert-publish-aot-testApp-warning-count.ps1:15]:
```suggestion
pushd $rootDirectory/test/OpenTelemetry.AotCompatibility.TestApp/bin/Debug/net7.0/linux-x64

```

(nit) I'd keep slashes consistent.

File: build/assert-publish-aot-testApp-warning-count.ps1 [hunk 3]
```diff
@@ -12,6 +12,20 @@ foreach ($line in $($publishOutput -split "`r`n"))
     }
 }
 
+pushd $rootDirectory/test/OpenTelemetry.AotCompatibility.TestApp/bin/Debug/net7.0/linux-x64
```
Comment (reyang) [inline build/assert-publish-aot-testApp-warning-count.ps1:15]: This is good for now, later we might need to find a way to remove the hard-coded "net7.0" version (by taking the version from environment variables).

Comment (reyang) [reply build/assert-publish-aot-testApp-warning-count.ps1:15]: For example, https://github.com/open-telemetry/opentelemetry-dotnet/blob/a831093ae18efeb7bcbcef1042d05c0cef05e7a8/.github/workflows/ci-instrumentation-libraries.yml#L41

**Review by reyang — APPROVED** [review]

**Review by eerhardt — APPROVED** [review]

File: .github/workflows/ci-aot.yml [hunk 4]
```diff
@@ -24,6 +24,6 @@ jobs:
       with:
         fetch-depth: 0 # fetching all
 
-    - name: Read publish AOT testApp output and assert warning count
+    - name: publish AOT testApp, assert static analysis warning count and exit code
```
Comment (utpilla) [inline .github/workflows/ci-aot.yml:27]:
```suggestion
    - name: publish AOT testApp, assert static analysis warning count, and run the app
```

File: build/assert-publish-aot-testApp-warning-count.ps1 (hunk 3 above)
Comment (utpilla) [inline build/assert-publish-aot-testApp-warning-count.ps1:15]: Be consistent with forward slash here since this test is only being run on Linux?

Update the command for dotnet publish:
`dotnet publish $rootDirectory\test\OpenTelemetry.AotCompatibility.TestApp\OpenTelemetry.AotCompatibility.TestApp.csproj -nodeReuse:false /p:UseSharedCompilation=false`

Comment (Yun-Ting) [reply build/assert-publish-aot-testApp-warning-count.ps1:15]: updated!

File: build/test-aot-compatibility.ps1 [hunk 5]
```diff
@@ -0,0 +1,38 @@
+$rootDirectory = Split-Path $PSScriptRoot -Parent
```
Comment (utpilla) [inline build/test-aot-compatibility.ps1:1]: Did you mistakenly add another Powershell script?

Comment (Yun-Ting) [reply build/test-aot-compatibility.ps1:3]: I wanted to rename the script. Fixed now.

**Review by utpilla — APPROVED** [review]
